src/bin/trace-verifier/commands/run_file.rs

```diff
@@ -1,9 +1,57 @@
-use crate::utils;
 use clap::Args;
 use eth_types::l2_types::BlockTrace;
+use serde::{Deserialize, Deserializer};
 use stateless_block_verifier::HardforkConfig;
 use std::path::PathBuf;
 
+use crate::utils;
+
+/// A wrapper around [`eth_types::l2_types::BlockTrace`] so that we can re-implement custom
+/// deserializer for it.
+struct WrappedBlockTrace(BlockTrace);
+
+/// In the JSON format of [`eth_types::l2_types::BlockTrace`] appears a "chainID" key.
+const JSON_KEY_CHAIN_ID: &str = "chainID";
+
+impl<'de> Deserialize<'de> for WrappedBlockTrace {
```

Comment:
Why this rather than the fallback method? This deserialization is not a hotspot and only for compatibility. The new trait impl is much hard to understand for others.

Reply:
Eventually are we guaranteed to use the "correct" JSON format for a trace? And disallow the JSON-RPC result JSON? I mean, was this fallback temporary?